### DISCUSSION TITLE
Private Key to CIP30 Wallet

### DISCUSSION THREAD

**Discussion by MartinSchere** [input-output-hk/cardano-js-sdk#1359] [Q&A]
Hello,

I am trying to generate a CIP30 interface from an ed25519 private key. Due to the lack of documentation, I'm having a hard time figuring out what pieces I need to put together.  I see there are things like Stores and Providers.

How could I go about doing this?

Thanks in advance

**Comment by rhyslbw** [comment]
The [CIP-30 to wallet mapping](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/wallet/src/cip30.ts) shows you how we're implementing the [component from the dapp-connector](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/dapp-connector/src/WalletApi/Cip30Wallet.ts) package. [Here](https://github.com/input-output-hk/cardano-js-sdk/blob/df77cab8b58bab2fb26a500fc592034087d13ee5/packages/e2e/src/factories.ts#L340-L422) a wallet is constructed for e2e tests. Please reach out again if needing further assistance
